lib/mongoid/matcher/bits_all_clear.rb

```diff
@@ -0,0 +1,24 @@
+module Mongoid
+  module Matcher
+
+    # @api private
+    module BitsAllClear
+      module_function def matches?(exists, value, condition)
+        # byebug
+        # p condition
+        case condition
+        when Array
+        #  array of bits
+        # TODO
+        when BSON::Binary
+          #condition is a binary string
+          # TODO: add logic to convert to integer
+        when Integer
+          value & condition
```

Comment:
I think this would need to be `value & condition > 0` or `value & condition == 0` depending on the desired condition as 0 is truthy in Ruby.